packages/next/lib/has-necessary-dependencies.ts

```diff
@@ -1,40 +1,23 @@
-import chalk from 'chalk'
-import { join } from 'path'
-
-import { fileExists } from './file-exists'
-import { getOxfordCommaList } from './oxford-comma-list'
-import { FatalError } from './fatal-error'
-
-const requiredTSPackages = [
-  { file: 'typescript', pkg: 'typescript' },
-  { file: '@types/react/index.d.ts', pkg: '@types/react' },
-  { file: '@types/node/index.d.ts', pkg: '@types/node' },
-]
-
-const requiredLintPackages = [
-  { file: 'eslint/lib/api.js', pkg: 'eslint' },
-  { file: 'eslint-config-next', pkg: 'eslint-config-next' },
-]
+export interface MissingDependency {
+  file: string
+  pkg: string
+}
 
 export type NecessaryDependencies = {
   resolved: Map<string, string>
+  missing: MissingDependency[]
 }
 
 export async function hasNecessaryDependencies(
   baseDir: string,
-  checkTSDeps: boolean,
-  checkESLintDeps: boolean,
-  lintDuringBuild: boolean = false
+  requiredPackages: MissingDependency[],
+  checkDeps: boolean = true
 ): Promise<NecessaryDependencies> {
-  if (!checkTSDeps && !checkESLintDeps) {
-    return { resolved: undefined! }
+  if (!checkDeps) {
+    return { resolved: undefined!, missing: [] }
```

Comment:
This should probably be `resolved: new Map()` so the consumers don't try to access a property on undefined

Reply:
Good call @LetItRock. Removed the prop altogether along with the return statement and just handled the logic in `verifyTypeScriptSetup.ts`.